includes/Modules/Analytics_4.php

```diff
@@ -445,15 +445,20 @@ private function register_tag() {
 			return;
 		}
 
-		$module_settings = $this->get_settings();
-		$settings        = $module_settings->get();
-		$tag             = new Web_Tag( $settings['measurementID'], self::MODULE_SLUG );
-		if ( $tag && ! $tag->is_tag_blocked() ) {
-			$tag->use_guard( new Tag_Verify_Guard( $this->context->input() ) );
-			$tag->use_guard( new Tag_Guard( $module_settings ) );
-			if ( $tag->can_register() ) {
-				$tag->register();
-			}
+		$settings = $this->get_settings()->get();
+		$tag      = new Web_Tag( $settings['measurementID'], self::MODULE_SLUG );
+
+		if ( $tag->is_tag_blocked() ) {
+			return;
+		}
+
+		$tag->use_guard( new Tag_Verify_Guard( $this->context->input() ) );
+		$tag->use_guard( new Tag_Guard( $this->get_settings() ) );
+
+		if ( $tag->can_register() ) {
+			$tag->set_ads_conversion_id( $settings['adsConversionID'] );
```

Comment:
See my other comment, this needs to come from the `Analytics\Settings` class instead. Since it won't be needed forever, let's not try to find a nice way of dependency injection here. What would be good to have though (also temporarily) is a private method like `get_analytics_module_settings()`, because that can also be used if we need to use the account ID from the `analytics` module in this class.

That method implementation could probably just manually instantiate the `Analytics\Settings` class and then call `get` on it.

Reply:
I think we can handle this in a more centralized way by proxying a few settings values from the classic settings rather than adding inline exceptions across the codebase. This way, the `Settings` class is responsible for this and the usage need not change later when we decouple them. I'll add this to the PR and we can revert it if you'd prefer not to, but I think it should make for an easier transition.

Reply:
Updated to come from Analytics settings for now 👍